app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -218,10 +219,32 @@ public static boolean HandleGoodReading(String tagId, byte[] data1, final long C
         return HandleGoodReading(tagId, data1, CaptureDateTime, false, null, null);
     }
 
+
+    public static void SendLibrereading(final String tagId, byte[] data1, final long CaptureDateTime, byte []patchUid,  byte []patchInfo){
+        if(!Home.get_master()) {
+            return;
+        }
+        LibreBlock libreBlock = LibreBlock.getForTimestamp(CaptureDateTime);
+        if (libreBlock != null) {
+            // We already have this one, so we have already sent it, so let's not crate storms.
+            return;
+        }
+        // Create the object to send
+        libreBlock = LibreBlock.create(tagId, CaptureDateTime, data1, 0, patchUid, patchInfo);
+        if(libreBlock == null) {
+            Log.e(TAG, "Error could not create libreBlock for libre-allhouse");
+            return;
+        }
+        final String json = libreBlock.toExtendedJson();
+        
+        GcmActivity.pushLibreBlock(json);
```

Comment:
This should be wrapped either within the function or the caller with both a preference switch that it only becomes active when libre all house is in use and a rate limit. It looks like you have neither from reading the PR?

Reply:
Libreblocks have the data to needed to show libretrend, see sensor age, and device battery.
This information should pass to normal followers as well. So I did not limit this on a preference swithc.

In the function pushLibreBlock() we have the rate limiter.